### ISSUE TITLE
The Window x86 and x64 Release legs in the dotnet-coreclr official are failing 

### ISSUE THREAD

**Issue by mikem8361** [dotnet/coreclr#25233]
The Window x86 and x64 Release legs in the dotnet-coreclr official are failing with this:

```
Running: dotnet msbuild "F:\workspace\_work\1\s\src\.nuget\optdata\ibcmerge.csproj" /t:DumpIbcMergePackageVersion /nologo /p:__BuildOS=Windows_NT /p:__BuildType=Release /p:__BuildArch=x64 /p:RestoreDuringBuild=true /p:OfficialBuildId=20190617.5 /p:IbcMergePackageVersionOutputFile=
F:\workspace\_work\1\s\src\.nuget\optdata\ibcmerge.csproj(24,5): error : IbcMergePackageVersionOutputFile must be passed as a property.
"Failed to determine IBC Merge version."
##[error]Cmd.exe exited with code '1'.
##[section]Finishing: Build product
```

https://dev.azure.com/dnceng/internal/_build/results?buildId=228006

I (with Juan's help) fixed a few delayed expansion problems in build.cmd but the ibcmerge needs to be installed/restored.

```
Running: dotnet msbuild "F:\workspace\_work\1\s\src\.nuget\optdata\ibcmerge.csproj" /t:DumpIbcMergePackageVersion /nologo /p:__BuildOS=Windows_NT /p:__BuildType=Release /p:__BuildArch=x64 /p:RestoreDuringBuild=true /p:OfficialBuildId=20190617.9 /p:IbcMergePackageVersionOutputFile=""F:\workspace\_work\1\s\bin\obj\Windows_NT.x64.Release\ibcmergeversion.txt""
  IbcMergePackageVersion: 5.0.6-beta.19203.1 written to: F:\workspace\_work\1\s\bin\obj\Windows_NT.x64.Release\ibcmergeversion.txt
Could not find IBCMerge at F:\workspace\_work\1\s\.packages\microsoft.dotnet.ibcmerge\99.99.99-master-20190614.1\tools\netcoreapp2.0\ibcmerge.dll. Have you restored src/.nuget/optdata/ibcmerge.csproj?
##[error]Cmd.exe exited with code '1'.
```

build.cmd changes;

```
diff --git a/build.cmd b/build.cmd
index f3090e7f79..1f6fbebfbb 100644
--- a/build.cmd
+++ b/build.cmd
@@ -659,18 +659,18 @@ if %__BuildCoreLib% EQU 1 (
         echo %__MsgPrefix%Commencing IBCMerge of System.Private.CoreLib for %__BuildOS%.%__BuildArch%.%__BuildType%
         set IbcMergeProjectFilePath=%__ProjectDir%\src\.nuget\optdata\ibcmerge.csproj
         set IbcMergePackageVersionOutputFile="%__IntermediatesDir%\ibcmergeversion.txt"
-        call "%__ProjectDir%\dotnet.cmd" msbuild "!IbcMergeProjectFilePath!" /t:DumpIbcMergePackageVersion /nologo %__CommonMSBuildArgs% /p:IbcMergePackageVersionOutputFile=%IbcMergePackageVersionOutputFile%
+        call "%__ProjectDir%\dotnet.cmd" msbuild "!IbcMergeProjectFilePath!" /t:DumpIbcMergePackageVersion /nologo %__CommonMSBuildArgs% /p:IbcMergePackageVersionOutputFile="!IbcMergePackageVersionOutputFile!"

         if not !errorlevel! == 0 (
             echo "Failed to determine IBC Merge version."
             exit /b !errorlevel!
         )
-        if not exist "%IbcMergePackageVersionOutputFile%" (
+        if not exist "!IbcMergePackageVersionOutputFile!" (
             echo "Failed to determine IBC Merge version."
             exit /b 1
         )

-        set /p __IbcMergeVersion=<"%IbcMergePackageVersionOutputFile%"
+        set /p __IbcMergeVersion=<"!IbcMergePackageVersionOutputFile!"

         set IbcMergePath=%__PackagesDir%\microsoft.dotnet.ibcmerge\!__IbcMergeVersion!\tools\netcoreapp2.0\ibcmerge.dll
         if exist !IbcMergePath! (
```

**Comment by hoyosjs** [comment]
I think the fix is here https://github.com/dotnet/coreclr/blob/d006ed403b942d42114445c23699af1ebdba2662/src/.nuget/optdata/ibcmerge.csproj#L9-L10 and here https://github.com/dotnet/coreclr/blob/d006ed403b942d42114445c23699af1ebdba2662/src/.nuget/optdata/ibcmerge.csproj#L27-29

The tool version is `IbcMergePackageVersion`, but we are writing `optimizationIBCCoreCLRVersion`. The main problem is both `__IbcMergeVersion` and `__IbcOptDataVersion` are populated from `IbcDataPackageVersionOutputFile`

**Comment by jashook** [comment]
There are a few issues, will put out a pr

**Comment by jashook** [comment]
Fixed with #25234
